src/components/DNSMap/DNSMap.tsx

```diff
@@ -30,14 +19,39 @@ interface FormValuesType {
     ipsToIgnore: string;
 }
 
-const DNSMap = () => {
-    const [loading, setLoading] = useState(false);
-    const [output, setOutput] = useState("");
-    const [checkedAdvanced, setCheckedAdvanced] = useState(false);
-    const [Pid, setPid] = useState("");
-    const [allowSave, setAllowSave] = useState(false);
-    const [hasSaved, setHasSaved] = useState(false);
+/**
+ * The DNSMap component.
+ * @returns The DNSMap component.
+ */
 
+const DNSMap = () => {
+    //Componenets state Variable
+    const [loading, setLoading] = useState(false); // State variable that represents loading state of the component
+    const [output, setOutput] = useState(""); // State variable that represents the output text from the DNS mapping process
+    const [checkedAdvanced, setCheckedAdvanced] = useState(false); //State variable that represents the state of advanced mode switch
+    const [Pid, setPid] = useState(""); //State variable that represents the process ID of the DNS mapping process
+    const [allowSave, setAllowSave] = useState(false); //State variable that represents whether saving the output is allowed
+    const [hasSaved, setHasSaved] = useState(false); //State variable that represents whether the output has been saved
+
+    //Components Constant Variables
+    const title = "DNS Mapping for Subdomains (DNSMap)";
+    const description_userguide =
+        "DNSMap scans a domain for common subdomains using a built-in or an external wordlist (if specified using -w option). " +
+        "The internal wordlist has around 1000 words in English and Spanish as ns1, firewall services and smtp. " +
+        "So it will be possible to search for smtp.example.com inside example.com automatically.\n\n";
+    const steps =
+        "Step 1: Enter a valid domain to be mapped.\n" +
+        "       Eg: google.com\n\n" +
+        "Step 2: Enter a delay between requests. Default is 10 (milliseconds). Can be left blank.\n" +
+        "       Eg: 10\n\n" +
+        "Step 3: Click 'Start Mapping' to commence the DNSMap tool's operation.\n\n" +
+        "Step 4: View the Output block below to view the results of the tool's execution.\n\n" +
+        "Switch to Advanced Mode for further options.";
+    const sourceLink = "https://www.kali.org/tools/dnsmap/"; // Link to the source code (or Kali Tools).
+    const tutorial = ""; // Link to the official documentation/tutorial.
+    const dependencies = ["DNSMap"]; // Contains the dependencies required by the component.
```

Comment:
Should be all lowercase. Needs to match the command entered into the terminal. 
```suggestion
    const dependencies = ["dnsmap"]; // Contains the dependencies required by the component.
```